octobot_trading/personal_data/portfolios/history/historical_asset_value.py

```diff
@@ -0,0 +1,67 @@
+#  Drakkar-Software OctoBot-Trading
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import decimal
+import copy
+
+
+class HistoricalAssetValue:
+    """
+    HistoricalAssetValue stores the portfolio value at a given time in multiple currencies
+    """
+    TIMESTAMP_KEY = "t"
+    VALUES_KEY = "v"
+
+    def __init__(self, timestamp, value_by_currency):
+        self._timestamp = timestamp
+        self._value_by_currency = copy.copy(value_by_currency)
+
+    def __contains__(self, item):
+        return item in self._value_by_currency
+
+    def get(self, currency):
+        return self._value_by_currency[currency]
+
+    def set(self, currency, value):
+        self._value_by_currency[currency] = value
```

Comment:
Should we update the timestamp if we are modifying the value?

Reply:
We could add a parameter for it. I didn't do it as there is no use for it, if we want to change something, it will always be the values or we will just delete the historicalAssetValue altogether. As of now, we don't need to change an asset's time (if we want to register values for another time, we just create a new historical asset)